mesos-client/src/main/scala/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -354,200 +135,276 @@ trait MesosApi {
 
-  /**
-    * Sent by the scheduler to explicitly decline offer(s) received. Note that this is same as sending an ACCEPT
-    * call with no operations.
-    *
-    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#decline
-    */
-  def decline(offerIds: Seq[OfferID], filters: Option[Filters] = None): Call = {
-    Call(
-      `type` = Some(Call.Type.DECLINE),
-      decline = Some(Decline(offerIds = offerIds, filters = filters))
-    )
-  }
+  private val eventDeserializer: Flow[ByteString, Event, NotUsed] =
+    Flow[ByteString].map(bytes => Event.parseFrom(bytes.toArray))
 
-  /**
-    * Sent by the scheduler to remove any/all filters that it has previously set via ACCEPT or DECLINE calls.
-    *
-    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#revive
-    */
-  def revive(role: Option[String] = None): Call = {
-    Call(
-      `type` = Some(Call.Type.REVIVE),
-      revive = Some(Revive(role = role))
-    )
-  }
+  private def connectionSource(frameworkInfo: FrameworkInfo, url: URI)(implicit mat: Materializer, as: ActorSystem) = {
+    val body = subscribe(frameworkInfo).toByteArray
 
-  /**
-    * Suppress offers for the specified roles. If `roles` is empty the `SUPPRESS` call will suppress offers for all
-    * of the roles the framework is currently subscribed to.
-    *
-    * http://mesos.apache.org/documentation/latest/upgrades/#1-2-x-revive-suppress
-    */
-  def suppress(roles: Option[String] = None): Call = {
-    Call(
-      `type` = Some(Call.Type.SUPPRESS),
-      suppress = Some(Call.Suppress(roles))
-    )
-  }
+    val request = HttpRequest(
+      HttpMethods.POST,
+      uri = Uri("/api/v1/scheduler"),
+      entity = HttpEntity(ProtobufMediaType, body),
+      headers = List(headers.Accept(ProtobufMediaType)))
 
-  /**
-    * Sent by the scheduler to kill a specific task. If the scheduler has a custom executor, the kill is forwarded
-    * to the executor; it is up to the executor to kill the task and send a TASK_KILLED (or TASK_FAILED) update.
-    * If the task hasn’t yet been delivered to the executor when Mesos master or agent receives the kill request,
-    * a TASK_KILLED is generated and the task launch is not forwarded to the executor. Note that if the task belongs
-    * to a task group, killing of one task results in all tasks in the task group being killed. Mesos releases the
-    * resources for a task once it receives a terminal update for the task. If the task is unknown to the master,
-    * a TASK_LOST will be generated.
-    *
-    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#kill
-    */
-  def kill(taskId: TaskID, agentId: Option[AgentID] = None, killPolicy: Option[KillPolicy]): Call = {
-    Call(
-      `type` = Some(Call.Type.KILL),
-      kill = Some(Kill(taskId = taskId, agentId = agentId, killPolicy = killPolicy))
-    )
+    val httpConnection = Http().outgoingConnection(url.getHost, url.getPort)
+    Source.single(request)
```

Comment:
Nit: A a new line above. I thought this was a continuation of `val httpConnection`.

Reply:
✅